routers/routes/routes.go

```diff
@@ -45,12 +46,34 @@ import (
 	macaron "gopkg.in/macaron.v1"
 )
 
+func giteaLogger(l *log.LoggerAsWriter) macaron.Handler {
+	return func(ctx *macaron.Context) {
+		start := time.Now()
+
+		l.Log(fmt.Sprintf("[Macaron] Started %s %s for %s", ctx.Req.Method, ctx.Req.RequestURI, ctx.RemoteAddr()))
+
+		rw := ctx.Resp.(macaron.ResponseWriter)
+		ctx.Next()
+
+		l.Log(fmt.Sprintf("[Macaron] Completed %s %s %v %s in %v", ctx.Req.Method, ctx.Req.RequestURI, rw.Status(), http.StatusText(rw.Status()), time.Since(start)))
+	}
+}
+
 // NewMacaron initializes Macaron instance.
 func NewMacaron() *macaron.Macaron {
 	gob.Register(&u2f.Challenge{})
-	m := macaron.New()
-	if !setting.DisableRouterLog {
-		m.Use(macaron.Logger())
+	var m *macaron.Macaron
+	if setting.RedirectMacaronLog {
+		loggerAsWriter := log.NewLoggerAsWriter("INFO")
```

Comment:
why the log level is `INFO` ?

Reply:
There are two places that we should discuss:

* Macaron's general log.
* The router log.

Macaron's log doesn't really differentiate log levels, it uses the go log framework and so differentiates info versus fatal simply by killing the process. It's difficult to choose a log level therefore if we're going to hijack this and send it elsewhere. We can't be sure that everything Macaron logs is an error not is it just debug - therefore `INFO` seems a good compromise. 

Regarding the router log, here we can actually be a bit better - we can choose the log level with the proposed GiteaLogger and use the Gitea log framework. In terms of choosing the log level though, it's difficult - we can turn these off specifically through its own setting and previously they were being spewed out to Stdout so again I thought `INFO` was a reasonable level.

I'm happy to change it.

Reply:
I like the idea of keeping it `INFO` because it is router logs, and that I think can be verbose.